pkg/wormhole/server.go

```diff
@@ -37,9 +35,6 @@ type Server struct {
 
 func NewServer(options *ServerOptions) (*Server, error) {
 	s := &Server{Logger: log.With(options.Logger, "wormhole", "server")}
-	if options.TunnelOptions.Logger == nil {
```

Comment:
Isn't the logger in TunnelOptions now `nil` again?

Reply:
No because I removed the logger from the our ServerOptions struct.
But while looking at the code I got dissatisfied with it. I think it was kind of convoluted due to fact that we split the work on it you writing the controller and my writing the tunnel server.
I'm taking the opportunity to refactor the wormhole server a bit to de-obfiuscate the options handling a little bit.